Source/DafnyDriver/CliCompilation.cs

```diff
@@ -228,4 +234,28 @@ public class CliCompilation {
       // Failed to resolve the program due to a user error.
     }
   }
+
+  private List<ICanVerify> FilterCanVerifies(List<ICanVerify> canVerifies) {
+    var filterPosition = options.Get(VerifyCommand.FilterPosition);
+    if (filterPosition == null) {
+      return canVerifies;
+    }
+
+    var regex = new Regex(@"(.+)(?::(\d+))?", RegexOptions.RightToLeft);
+    var result = regex.Match(filterPosition);
+    if (result.Length != filterPosition.Length || !result.Success) {
+      Compilation.Reporter.Error(MessageSource.Project, Token.Cli, "Could not parse value passed to --filter-position");
+      return new List<ICanVerify>();
+    }
+    var filePart = result.Groups[1].Value;
+    string? linePart = result.Groups.Count > 2 ? result.Groups[2].Value : null;
+    var fileFiltered = canVerifies.Where(c => c.Tok.Uri.ToString().EndsWith(filePart)).ToList();
+    if (string.IsNullOrEmpty(linePart)) {
+      return fileFiltered;
+    }
+
+    var line = int.Parse(linePart);
+    return fileFiltered.Where(c =>
```

Comment:
To confirm: you are trusting that the delimited portion of the file is valid standalone Dafny code?

Reply:
No, the entire Dafny project is parsed and resolved regardless. After resolution the verifiable parts are extracted, and this filter is applied to those to select what exactly to verify.